code-dot-org/code-dot-org
# import CSA standards framework into code studio

Finishes [PLAT-942]. The task was to create a generalized script for importing standards. However, we don't know what format future standards frameworks will be in, so I felt a bit more comfortable including "csa" in the name of this import script since this is the only framework we know it will work for. If this input format works well for other future standards frameworks, then we could rename the script to be more general.

The original gsheet this data came from is here: [CSA Curriculum Development Tracker](https://docs.google.com/spreadsheets/d/1lVecnUAi2DjQcwdqxf7Gry_KD4CuYDkrzMJCAnhI51A/edit#gid=691314790) --> Frameworks

For easier viewing, this gsheet shows the input data alongside the output data, which are the result of importing the _categories.csv and _standards.csv files: https://docs.google.com/spreadsheets/d/18r6kEuykgjSfDhJvqbcs6guwItx8uQxBqrvR7OHgqvg/edit#gid=691314790 

## Testing story

manually verified as follows:
* ran `split-csa-standards.rb csa.csv` against [csa.csv.zip](https://github.com/code-dot-org/code-dot-org/files/6282324/csa.csv.zip) to generate the _standards.csv and _categories.csv files seen in this PR
* ran `rake seed:standards`
* verified in rails console that the correct number of standards and categories now exist in the DB for this new framework
* verified in lesson editor that this framework shows up and that standards in it can be added to lessons:

in order to get the last step to work, I had to patch in #39972 to make the new framework appear in the dropdown.

[PLAT-942]: https://codedotorg.atlassian.net/browse/PLAT-942
